packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -273,4 +272,18 @@ class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
         ))
         handler.post { flutterResult.error("AmplifyException", msg, errorMap) }
     }
+
+    @VisibleForTesting
+    fun deserializeNestedModels(serializedModelData: Map<String, Any>): Map<String, Any> {
+        return serializedModelData.mapValues {
+            if(it.value is Map<*, *>) {
```

Comment:
Should this not check that the field is actually a `model` based on the schema?

Reply:
I don't believe we support Maps as a field type in `Model` so this is pretty safe assumption to make.